backend/ops_api/ops/resources/agreements.py

```diff
@@ -120,6 +154,109 @@ def get(self, id: int) -> Response:
 
         return response
 
+    @override
+    @jwt_required()
+    def put(self, id: int) -> Response:
+        message_prefix = f"PUT to {ENDPOINT_STRING}"
+
+        identity = get_jwt_identity()
+        is_authorized = self.auth_gateway.is_authorized(identity, ["PUT_AGREEMENT"])
+        if not is_authorized:
+            return make_response_with_headers({}, 401)
+
+        try:
+            with OpsEventHandler(OpsEventType.UPDATE_AGREEMENT) as meta:
+                old_agreement: Agreement = self._get_item(id)
+                if not old_agreement:
+                    raise RuntimeError("Invalid Agreement id.")
+                # reject change of agreement_type
+                # for PUT, it must exist in request
+                req_type = request.json.get("agreement_type", None)
+                if req_type != old_agreement.agreement_type.name:
+                    msg = "Invalid agreement_type, agreement_type must not change"
+                    current_app.logger.error(f"{message_prefix}: {msg}")
```

Comment:
to remove duplication here consider just throwing a RuntimeError as done above

Reply:
done